tests/virt-handler_test.go

```diff
@@ -0,0 +1,118 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors
+ *
+ */
+
+package tests_test
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	kvcorev1 "kubevirt.io/client-go/kubevirt/typed/core/v1"
+
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/libnode"
+	"kubevirt.io/kubevirt/tests/libvmops"
+
+	virt_api "kubevirt.io/kubevirt/pkg/virt-api"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/tests/libnet"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+)
+
+var _ = Describe("[sig-compute]virt-handler", decorators.SigCompute, func() {
+	// Regression
+	It("[Serial] multiple HTTP calls should re-use connections and not grow the number of open connections", Serial, func() {
+
+		getHandlerConnectionCount := func(nodeName string) int {
+			cmd := []string{"bash", "-c", fmt.Sprintf("ss -ntlap | grep %d | wc -l", virt_api.DefaultConsoleServerPort)}
+			stdout, stderr, err := libnode.ExecuteCommandOnNodeThroughVirtHandler(nodeName, cmd)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(stderr).To(BeEmpty())
+
+			stdout = strings.TrimSpace(stdout)
+			stdout = strings.ReplaceAll(stdout, "\n", "")
+
+			handlerCons, err := strconv.Atoi(stdout)
+			Expect(err).ToNot(HaveOccurred())
+
+			return handlerCons
+		}
+
+		getClientCalls := func(vmi *v1.VirtualMachineInstance) []func() {
+			vmiInterface := kubevirt.Client().VirtualMachineInstance(vmi.Namespace)
+			expectNoErr := func(err error) {
+				ExpectWithOffset(2, err).ToNot(HaveOccurred())
+			}
+
+			return []func(){
+				func() {
+					_, err := vmiInterface.GuestOsInfo(context.Background(), vmi.Name)
+					expectNoErr(err)
+				},
+				func() {
+					_, err := vmiInterface.FilesystemList(context.Background(), vmi.Name)
+					expectNoErr(err)
+				},
+				func() {
+					_, err := vmiInterface.UserList(context.Background(), vmi.Name)
+					expectNoErr(err)
+				},
+				func() {
+					_, err := vmiInterface.VNC(vmi.Name)
+					expectNoErr(err)
+				},
+				func() {
+					_, err := vmiInterface.SerialConsole(vmi.Name, &kvcorev1.SerialConsoleOptions{ConnectionTimeout: 30 * time.Second})
+					expectNoErr(err)
+				},
+			}
```

Comment:
Assign this directly to `clientCalls` instead of wrapping it in a lambda.

Reply:
This is just moved from one place (which didn't make sense to have it) to more proper one. I am not sure I will be able to follow up on this in short term.